### PR TITLE
Disallow explicit PASE in access control

### PR DESCRIPTION
**Pull request by mlepage-google**
#### Problem
Although we never intended to support operational
PASE for v1.0, some plumbing still existed for it.
E.g. the module would still check against explicit
PASE entries, and this was unit tested, even though
the cluster would not allow them to be created in
production. This should be removed/disallowed for
v1.0 in favour of simplicity.

#### Change overview
We won't have explicit PASE ACL entries for v1.0.
We will enforce that PASE is only during
commissioning, therefore all incoming PASE subjects\
will be granted administer privilege.

Past v1.0, if/when we want operational PASE (requires
solving some tricky multi-fabric issues, so may not be
feasible), we'll have to check against explicit PASE
subjects in entries, and also for implicit PASE administer
privilege during commissioning we'll have to verify that
the incoming PASE subject is commissioning (otherwise
it should not get that implicit privilege escalation).

Part of issue #10242

#### Testing
- Unit tests pass (on Linux host)
- Ran all-clusters-app with REPL on Linux host, read attribute

### CODE REVIEW

**Comment by mlepage-google** [comment]
Unit tests updated to match revised code, should now pass.

File: src/access/AccessControl.cpp [hunk 1]
```diff
@@ -96,6 +99,8 @@ CHIP_ERROR AccessControl::Check(const SubjectDescriptor & subjectDescriptor, con
     {
         AuthMode authMode = AuthMode::kNone;
         ReturnErrorOnFailure(entry.GetAuthMode(authMode));
+        // Operational PASE not supported for v1.0.
+        VerifyOrReturnError(authMode == AuthMode::kCase || authMode == AuthMode::kGroup, CHIP_ERROR_INCORRECT_STATE);
```
Comment (tcarmelveilleux) [inline src/access/AccessControl.cpp:103]: kGroup is valid. Why is kGroup an incorrect state at this point?

**Review by tcarmelveilleux — APPROVED** [review]

File: src/access/AccessControl.cpp (hunk 1 above)
Comment (mlepage-google) [reply src/access/AccessControl.cpp:103]: It's OK, that line is doing:
verify that authmode is case or group, else return error.

Comment (tcarmelveilleux) [reply src/access/AccessControl.cpp:103]: Got fooled again by reverse logic macro steering my eyes

Comment (mlepage-google) [reply src/access/AccessControl.cpp:103]: :-)

**Review by msandstedt — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**